bitnami/tomcat/templates/_pod.tpl

```diff
@@ -150,6 +150,24 @@ containers:
     volumeMounts:
       - name: data
         mountPath: /bitnami/tomcat
+      - name: empty-dir
+        mountPath: /opt/bitnami/tomcat/webapps
+        subPath: app-webapps-dir
```

Comment:
`/opt/bitnami/tomcat/webapps` is "virtually mounted" and persisted in the `/bitnami/tomcat` volume. If we mount this empty volume, we will remove installed apps every time the pod is removed.

Reply:
Actually, we're not persisting the webapps, looking at the container logic. What we're doing in the container logic is checking if the webapps folder is empty or not, but we're not applying any persistence logic to that folder.

```
    # Fix to make upgrades from old images work
    # Before, we were persisting 'data' dir instead of 'webapps', causing errors when restoring persisted data
    if ! is_dir_empty "$TOMCAT_WEBAPPS_DIR" || ! is_dir_empty "${TOMCAT_VOLUME_DIR}/data"; then
        info "Persisted webapps detected"
        if [[ ! -e "$TOMCAT_WEBAPPS_DIR" && -e "${TOMCAT_VOLUME_DIR}/data" ]]; then
            warn "Detected legacy configuration directory path ${TOMCAT_VOLUME_DIR}/conf in volume"
            warn "Creating ${TOMCAT_BASE_DIR}/webapps symlink pointing to ${TOMCAT_VOLUME_DIR}/data"
            ln -sf "${TOMCAT_VOLUME_DIR}/data" "${TOMCAT_BASE_DIR}/webapps"
        fi
    else
        info "Ensuring Tomcat directories exist"
        ensure_dir_exists "$TOMCAT_WEBAPPS_DIR"
        # Use tomcat:root ownership for compatibility when running as a non-root user
        am_i_root && configure_permissions_ownership "$TOMCAT_WEBAPPS_DIR" -d "775" -f "664" -u "$TOMCAT_DAEMON_USER" -g "root"

        if is_boolean_yes "$TOMCAT_INSTALL_DEFAULT_WEBAPPS"; then
            info "Deploying Tomcat from scratch"
            cp -rp "$TOMCAT_BASE_DIR"/webapps_default/* "$TOMCAT_WEBAPPS_DIR"

            # These applications have been enabled for historical reasons, and do not pose any security threat
            tomcat_enable_application examples
            tomcat_enable_application docs
            if is_boolean_yes "$TOMCAT_ALLOW_REMOTE_MANAGEMENT"; then
                # These applications should not be enabled by default, for security reasons
                info "Enabling remote connections for manager and host-manager applications"
                tomcat_enable_application manager
                tomcat_enable_application host-manager
            fi
        else
            info "Skipping deployment of default webapps"
        fi

    fi
```

However, we should check in an init container if the container has populated the webapps directory.

Reply:
I think I am missing something. In the postunpack.sh file: https://github.com/bitnami/containers/blob/7fa47487ccc2519fdfeffdff630d0ed38cbc3c71/bitnami/tomcat/10.1/debian-12/rootfs/opt/bitnami/scripts/tomcat/postunpack.sh#L37-L39

We can see that `${TOMCAT_BASE_DIR}/webapps` is a symlink to `$TOMCAT_WEBAPPS_DIR` (by default `/bitnami/tomcat/webapps`). If we mount an emptyDir over `${TOMCAT_BASE_DIR}/webapps` we will need to recreate the logic above in the chart.